### PR TITLE
Fix build and sign GitHub release

### PR DESCRIPTION
**Pull request by james-garriss**
## 🗣 Description ##

The build and sign GitHub release workflow uses some of the same publish code that the publish to PSGallery workflows use (see utils/workflow/Publish-ScubaGear.ps1).  The publish code was recently refactored, but the GitHub workflow was not updated to use the changes.  This PR corrects that oversight.

## 💭 Motivation and context ##

Closes: #1255 
Future improvements: #1257 

## 🧪 Testing ##

Checked the workflow pipeline.
Published an RC to GitHub.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] Changes are sized such that they do not touch excessive number of files.
- [x] These code changes follow the ScubaGear [content style guide]
- [x] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant repo and/or project documentation updated to reflect these changes.
- [x] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [x] PR passed smoke test check.
- [x] Feature branch has been rebased against changes from parent branch, as needed
- [x] Resolved all merge conflicts on branch
- [x] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

File: .github/workflows/build_sign_release.yaml [hunk 1]
```diff
@@ -32,42 +32,60 @@ jobs:
...
-            Remove-Item -Recurse -Force repo -Include .git*
-            SignScubaGearModule `
-              -AzureKeyVaultUrl $($KeyVaultInfo.KeyVault.URL) `
-              -CertificateName $($KeyVaultInfo.KeyVault.CertificateName) `
-              -ModulePath 'repo'
-            Move-Item  -Path repo -Destination "ScubaGear-${env:RELEASE_VERSION}" -Force
-            Compress-Archive -Path "ScubaGear-${env:RELEASE_VERSION}" -DestinationPath "ScubaGear-${env:RELEASE_VERSION}.zip"
-          azPSVersion: "latest"
-      - name: release
+        run: |
```
Comment (dagarwal-mitre) [inline .github/workflows/build_sign_release.yaml:61]: For this run from line 61-64, shouldn't there a name associated with this run or is it all part of getting the get key vault info?

**Review by dagarwal-mitre — APPROVED** [review]
Check 1 question I had

File: .github/workflows/build_sign_release.yaml (hunk 1 above)
Comment (james-garriss) [reply .github/workflows/build_sign_release.yaml:61]: There are 2 jobs here, starting with lines 57 and 65, and both have their own names.  Probably hard to see this from just looking at the change page on GitHub, as it often chops things into little pieces.

**Comment by james-garriss** [comment]
@nanda-katikaneni , ready to merge, urgent priority!

**Pull request merged**